sourcegraph/sourcegraph
# sg lint: fix 'buf lint' rule by actually adding buf.yaml file

When I was refactoring https://github.com/sourcegraph/sourcegraph/pull/47037, I accidentally `git` removed the `buf.yaml` file that `sg lint protobuf` relies on to discover which directories it run `buf` against. Ultimate, this caused `buf lint` to never run in CI (since the `buf.yaml` file was never added). 

This PR fixes this by:

- `git add`-ing the `internal/buf.yaml` file. 
- forcing `sg protobuf lint` to explicitly error if it fails to discover any configuration files to run `buf against` (as this is 99% not the author's intention)


This PR also fixes the failing `buf lint` rules by adjusting gitserver's enum naming in accordance with https://docs.buf.build/lint/rules#enum_value_prefix

## Test plan



Manual testing:

- run `git clean -dxf`
- `git checkout 3cc827ad25e6ae02ffc7c84d94174316df77815a`: (before the lint fixes)
- run `go run ./dev/sg lint --sfc protobuf` and see the lint failure
- `git checkout 1a3c5c9561252e7673f323bbb7847e09d48782af`: (after the lint fixes)
- run `git clean -dxf`
- run `go run ./dev/sg lint --sfc protobuf` and see the lint pass
